### PR TITLE
Issue/3665 error handling

### PR DESCRIPTION
**Pull request by mzorz**
Addresses issue #3665 and shows a toast if there was an error retrieving/saving the My Profile data, with a specific text for `401`and a generic error text for any other error

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/MyProfileActivity.java [hunk 1]
```diff
@@ -174,4 +175,17 @@ public void onEventMainThread(PrefsEvents.MyProfileDetailsChanged event) {
             refreshDetails();
         }
     }
+
+    public void onEventMainThread(PrefsEvents.MyProfileDataLoadSaveError event) {
+        if (!isFinishing()) {
+            if (event != null
```
Comment (mzorz) [inline WordPress/src/main/java/org/wordpress/android/ui/prefs/MyProfileActivity.java]: not really sure about this null check (while we need to check each nested element, it looks a bit awkward).
Also, not sure if we should take care of the 401 case in a specific manner; I'm doing it like so because that's the specific issue I'm addressing. More cases (and specific error texts) can be added if needed.

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/MyProfileActivity.java [hunk 2]
```diff
@@ -174,4 +175,17 @@ public void onEventMainThread(PrefsEvents.MyProfileDetailsChanged event) {
...
+                    && event.mVolleyError != null
+                    && event.mVolleyError.networkResponse != null
+                    && event.mVolleyError.networkResponse.statusCode == 401) {
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/prefs/MyProfileActivity.java]: You can simplify this using [VolleyUtils.statusCodeFromVolleyError()](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/util/VolleyUtils.java#L34).

File: WordPress/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -935,6 +935,8 @@
     <string name="error_refresh_unauthorized_comments">You don\'t have permission to view or edit comments</string>
     <string name="error_refresh_unauthorized_pages">You don\'t have permission to view or edit pages</string>
     <string name="error_refresh_unauthorized_posts">You don\'t have permission to view or edit posts</string>
+    <string name="error_refresh_unauthorized_profile">You don\'t have permission to view or edit your profile - check your configured site is whitelisted</string>
```
Comment (nbradbury) [inline WordPress/src/main/res/values/strings.xml]: Can the user "check your configured site is whitelisted" from the app? If not, perhaps that part of the error message should be dropped (it's a bit long anyway).

File: WordPress/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -935,6 +935,8 @@
...
+    <string name="error_refresh_profile_not_loaded">There was an error trying to access your profile information</string>
```
Comment (nbradbury) [inline WordPress/src/main/res/values/strings.xml]: Can this be shortened? Something like "Can't access your profile information" would work.

File: WordPress/src/main/res/values/strings.xml (hunk 3 above)
Comment (maxme) [inline WordPress/src/main/res/values/strings.xml]: Yes please drop the `error_refresh_unauthorized_profile` case.

**Comment by mzorz** [comment]
`error_refresh_unauthorized_profile`   case dropped @maxme

File: WordPress/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -935,6 +935,7 @@
     <string name="error_refresh_unauthorized_comments">You don\'t have permission to view or edit comments</string>
     <string name="error_refresh_unauthorized_pages">You don\'t have permission to view or edit pages</string>
     <string name="error_refresh_unauthorized_posts">You don\'t have permission to view or edit posts</string>
+    <string name="error_refresh_profile_not_accessible">There was an error trying to access your profile information</string>
```
Comment (nbradbury) [inline WordPress/src/main/res/values/strings.xml]: As with my earlier comment, can this be shortened? "Unable to access your profile" works.

Comment (mzorz) [inline WordPress/src/main/res/values/strings.xml]: Sure! I'll shorten this to that text. Sorry, what was your earlier comment?

Comment (mzorz) [inline WordPress/src/main/res/values/strings.xml]: done :+1:

Comment (nbradbury) [inline WordPress/src/main/res/values/strings.xml]: Looks like it got collapsed due to an "outdated diff" - https://github.com/wordpress-mobile/WordPress-Android/pull/3698#discussion_r51639762

Comment (mzorz) [inline WordPress/src/main/res/values/strings.xml]: ah got it :) :+1: I left it with "Unable to access your profile" as per your last request

**Comment by nbradbury** [comment]
Looks good! :shipit:
